File: citaria_backend/src/main/java/com/citaria/model/HorarioEmpleado.java
package com.citaria.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;

import java.time.LocalTime;
import java.util.Objects;

/**
 * Disponibilidad del empleado por día.
 */
@Entity
@Table(name = "horario_empleado")
public class HorarioEmpleado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "empleado_id", nullable = false)
    private Empleado empleado;

    @Column(name = "dia_semana", nullable = false)
    private Integer diaSemana;

    @Column(name = "hora_inicio", nullable = false)
    private LocalTime horaInicio;

    @Column(name = "hora_fin", nullable = false)
    private LocalTime horaFin;

    @Column(nullable = false)
    private Boolean activo = true;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Empleado getEmpleado() {
        return empleado;
    }

    public void setEmpleado(Empleado empleado) {
        this.empleado = empleado;
    }

    public Integer getDiaSemana() {
        return diaSemana;
    }

    public void setDiaSemana(Integer diaSemana) {
        this.diaSemana = diaSemana;
    }

    public LocalTime getHoraInicio() {
        return horaInicio;
    }

    public void setHoraInicio(LocalTime horaInicio) {
        this.horaInicio = horaInicio;
    }

    public LocalTime getHoraFin() {
        return horaFin;
    }

    public void setHoraFin(LocalTime horaFin) {
        this.horaFin = horaFin;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof HorarioEmpleado that)) return false;
        return Objects.equals(id, that.id) && Objects.equals(empleado, that.empleado) && Objects.equals(diaSemana, that.diaSemana) && Objects.equals(horaInicio, that.horaInicio) && Objects.equals(horaFin, that.horaFin) && Objects.equals(activo, that.activo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, empleado, diaSemana, horaInicio, horaFin, activo);
    }

    @Override
    public String toString() {
        return "HorarioEmpleado{" +
                "id=" + id +
                ", empleadoId=" + (empleado != null ? empleado.getId() : null) +
                ", diaSemana=" + diaSemana +
                ", horaInicio=" + horaInicio +
                ", horaFin=" + horaFin +
                ", activo=" + activo +
                '}';
    }
}

File: citaria_frontend/lib/data/models/horario_empleado.dart
class HorarioEmpleado {
  final int? id;
  final int? empleadoId;
  final int diaSemana;
  final String horaInicio;
  final String horaFin;
  final bool activo;

  const HorarioEmpleado({
    this.id,
    this.empleadoId,
    required this.diaSemana,
    required this.horaInicio,
    required this.horaFin,
    required this.activo,
  });

  factory HorarioEmpleado.fromJson(Map<String, dynamic> json) {
    return HorarioEmpleado(
      id: json['id'] == null ? null : (json['id'] as num).toInt(),
      empleadoId: json['empleadoId'] == null
          ? null
          : (json['empleadoId'] as num).toInt(),
      diaSemana: (json['diaSemana'] as num).toInt(),
      horaInicio: json['horaInicio'] as String,
      horaFin: json['horaFin'] as String,
      activo: json['activo'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'diaSemana': diaSemana,
      'horaInicio': horaInicio,
      'horaFin': horaFin,
      'activo': activo,
    };
  }
}
